src/Controls/tests/DeviceTests/Elements/Modal/ModalTests.iOS.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Maui.Controls;
+using Microsoft.Maui.Graphics;
+using Microsoft.Maui.Handlers;
+using Xunit;
+
+namespace Microsoft.Maui.DeviceTests
+{
+	public partial class ModalTests : ControlsHandlerTestBase
+	{
+		[Theory]
+		[ClassData(typeof(PageTypes))]
+		public async Task PushModalUsingTransparencies(Page rootPage, Page modalPage)
+		{
+			SetupBuilder();
+
+			var expected = Colors.Red;
+
+			rootPage.BackgroundColor = expected;
+			modalPage.BackgroundColor = Colors.Transparent;
+
+			await CreateHandlerAndAddToWindow<IWindowHandler>(rootPage,
+				async (handler) =>
+				{
+					var currentPage = (rootPage as IPageContainer<Page>).CurrentPage;
+					await currentPage.Navigation.PushModalAsync(modalPage);
+					await OnLoadedAsync(modalPage);
+					Assert.Equal(1, currentPage.Navigation.ModalStack.Count);
+
+					await handler.PlatformView.AssertContainsColor(expected);
```

Comment:
I'm running this locally and it's taking forever to loop through the pages and find anything

I don't think you need to search for the color here. 

I'm pretty sure you just need to validate if the "rootPage" still has a UIWindow set on the platformview because what the code is doing is changing the presentation style on the modal so the underlying view isn't removed.